### PR TITLE
CLN: Add import check helper

### PR DESCRIPTION
**Pull request by skailasa**
Addresses #496. 

- Adds a function to handle unavailable imports, and test it.
- Remove boilerplate for these imports throughout tests.
- Remove unused imports where found.

### CODE REVIEW

**Comment by mdickinson** [comment]
@skailasa Thanks for this. It went through a couple of rounds of refactoring, and things emerged from that looking a bit different to how they went in.

@rahulporuri Would you be interested in reviewing?

File: traits/testing/tests/test_optional_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from traits.testing.optional_dependencies import optional_import
+
+
+class MockModule(ModuleType):
+    """Mock module as object for testing"""
+    pass
+
+
+# Assign to sys modules to allow import
+sys.modules['mock_module'] = MockModule(name='mock')
```
Comment (rahulporuri) [inline traits/testing/tests/test_optional_dependencies.py]: Do want to remove this from `sys.modules` after the test is complete? Maybe add it to `sys.modules` in the test setup and remove it in the teardown?

Comment (mdickinson) [reply traits/testing/tests/test_optional_dependencies.py]: Argh. Yes, definitely. I spotted this first time round and then forgot about it (and also forgot to comment on it).

Maybe it would be even better to use something that we know is importable, like one of the standard library modules.

File: traits/tests/test_get_traits.py [hunk 2]
```diff
@@ -17,13 +17,10 @@
 import unittest
 
 from traits.api import (
-    Float,
```
Comment (rahulporuri) [inline traits/tests/test_get_traits.py]: unrelated cleanup?

Comment (mdickinson) [reply traits/tests/test_get_traits.py]: Yes; let's move this out of this PR.

File: traits/tests/test_trait_get_set.py [hunk 3]
```diff
@@ -16,7 +16,6 @@
 import logging
 import unittest
 
-import traits.has_traits
```
Comment (rahulporuri) [inline traits/tests/test_trait_get_set.py]: unrelated cleanup?

File: traits/testing/optional_dependencies.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+requires_numpy = unittest.skipIf(numpy is None, "NumPy not available")
+
+sphinx = optional_import("sphinx")
+requires_sphinx = unittest.skipIf(sphinx is None, "Sphinx not available")
+
+traitsui = optional_import("traitsui")
+requires_traitsui = unittest.skipIf(traitsui is None, "TraitsUI not available")
+
+requires_python2 = unittest.skipUnless(six.PY2, "Applicable only to Python 2")
+conflicts_with_python2 = unittest.skipIf(six.PY2, "Does not apply to Python 2")
```
Comment (rahulporuri) [inline traits/testing/optional_dependencies.py]: not a fan of "conflicts_with_python2" but i can't come up with a different/better name.

File: traits/testing/tests/test_optional_dependencies.py (hunk 1 above)
Comment (rahulporuri) [reply traits/testing/tests/test_optional_dependencies.py]: replacing the existing code with a stdlib module works for me.

File: traits/tests/test_get_traits.py (hunk 2 above)
Comment (rahulporuri) [reply traits/tests/test_get_traits.py]: no, i'm fine with leaving them in. (i don't mind cleaning up the codebase in a piecemeal fashion). i just wanted to confirm that it's unrelated.

File: traits/testing/optional_dependencies.py (hunk 4 above)
Comment (mdickinson) [reply traits/testing/optional_dependencies.py]: `requires_python3` is inaccurate, since what we really want is Python 3 or later (and it might not actually be that long until that matters). We could do `requires_python3_or_later`, but it's a bit long, and the name doesn't really reflect the real meaning of the decorator, which is that the test doesn't work on Python 2.

File: traits/tests/test_get_traits.py (hunk 2 above)
Comment (mdickinson) [reply traits/tests/test_get_traits.py]: I do mind. :-) This doesn't really fall under the boy scout rule: it's not an area of the codebase that we're touching otherwise, so it shouldn't be in this PR. PRs should be focused on one thing.
